lib/rubocop/cop/lint/unreachable_code.rb

```diff
@@ -88,6 +106,33 @@ def check_case(node)
 
           branches.all? { |branch| branch.body && flow_expression?(branch.body) }
         end
+
+        def handle_def(node)
+          @redefined << node.method_name
+          false
+        end
```

Comment:
This should probably only add the method to `@redefined` if it's one of the flow control methods.